src/jit/earlyprop.cpp

```diff
@@ -307,72 +309,57 @@ bool Compiler::optEarlyPropRewriteTree(GenTreePtr tree)
             {
                 GenTreeBoundsChk* check = tree->gtNext->AsBoundsChk();
 
-                if ((check->gtArrLen == tree) && check->gtIndex->IsCnsIntOrI() &&
-                    (check->gtIndex->AsIntCon()->IconValue() >= 0) &&
-                    (check->gtIndex->AsIntCon()->IconValue() < actualVal->AsIntCon()->IconValue()))
+                if ((check->gtArrLen == tree) && check->gtIndex->IsCnsIntOrI())
```

Comment:
What are these changes for?

Reply:
To remove copy paste of `actualVal->AsIntCon()->IconValue()` and `check->gtIndex->AsIntCon()->IconValue()`.

Reply:
At the expense of adding new local variables and splitting the condition into multiple if statements. I don't think that such trivial expression can be considered "copy paste" to warrant this.

Reply:
This seems like a reasonable change to me.